src/useFieldArray.ts

```diff
@@ -244,7 +244,11 @@ export const useFieldArray = <
       readFormStateRef.current.isDirty
     ) {
       set(dirtyFields, name, [
-        ...get(dirtyFields, name, fillEmptyArray(fields)),
+        ...get(
+          dirtyFields,
+          name,
+          fillEmptyArray(new Array(allFields.current.length)),
```

Comment:
```suggestion
          fillEmptyArray(allFields.current),
```

hmm, honestly I am trying to move away with that ref object :(

Reply:
I am happy to get this PR over the line, but I can't guarantee future change by completely removing that ref object. 🙏

Reply:
The thing that it is still used here anyway - line 256 (260). That works to eliminate `fields` dep.
If you have better solution I'd be happy to see it :)